src/snarky.d.ts

```diff
@@ -51,489 +53,767 @@ declare interface ProvablePure<T> extends Provable<T> {
+   * ```
+   * 
+   * **Warning: This is a modular multiplication in the pasta field.
+   * 
+   * ```ts
+   * const value1 = Field(409034234915762252310977182347851274525699134624909316457281397679077977n);
+   * const value2 = Field(396799155229642886214345240347513675246314887144578160280866744058082566n);
+   * 
+   * const product = value1.mul(value2);
+   * 
+   * // If you try to print product - `console.log(product.toBigInt())` - you will realize that it prints a very big integer, so you cannot use arithmetic operations on snarkyJS as in regular programming languages.
+   * // However, all the operations always stay provable: This means you can use the reverse operation of multiplication (division) to prove the product is calculated correctly.
+   * product.div(value1).assertEquals(value2);
+   * product.div(value2).assertEquals(value1);
+   * ```
+   * 
+   * @param value - a `field-like` value to multiply with the {@link Field}.
+   * 
+   * @return A {@link Field} element equivalent to the modular difference of the two value.
    */
-  mul(y: Field | number | string | boolean): Field;
+  mul(value: Field | number | string | boolean): Field;
 
   /**
-   * Divides this {@link Field} element through another coercible to a field.
+   * Divide another `field-like` element through this {@link Field} and returns the result.
+   *
+   * ```ts
+   * const value1 = Field(6);
+   * const value2 = 3; // A `field-like` element
+   * 
+   * const product = value1.div(value2);
+   * 
+   * product.assertEquals(Field(2));
+   * ```
+   * 
+   * **Warning: This is a modular division in the pasta field. You can think it as the reverse operation of modular multiplication.
+   * 
+   * ```ts
+   * const value1 = Field(2);
+   * const value2 = Field(5);
+   * 
+   * const quotient = value1.div(value2);
+   * 
+   * // If you try to print product - `console.log(quotient.toBigInt())` - you will realize that it prints a very big integer, so you cannot use arithmetic operations on snarkyJS as in regular programming languages.
+   * // However, all the operations always stay provable: This means you can use the reverse operation of division (multiplication) to prove the quotient is calculated correctly.
+   * quotient.mul(value2).assertEquals(value1);
+   * ```
+   * 
+   * @param value - a `field-like` value to divide with the {@link Field}.
+   * 
+   * @return A {@link Field} element equivalent to the modular division of the two value.
    */
-  div(y: Field | number | string | boolean): Field;
+  div(value: Field | number | string | boolean): Field;
 
   /**
-   * Squares this {@link Field} element.
+   * Square this {@link Field} element.
    *
    * ```typescript
-   * const x2 = x.square();
-   * x2.assertEquals(x.mul(x));
+   * const someField = Field(7);
+   * const square = someField.square();
+   * 
+   * square.assertEquals(someField.mul(someField)); // This statement is always true regardless of the value of `someField`
```

Comment:
I'm really not sure about changes like this, where everything is made more verbose and comments are added. To me, the original comment is cleaner, and therefore easier to understand.

changing `x` to `someField`, and showing the `Field(7)` construction, and the long comment all just add visual noise IMO

Since  this is done throughout this PR, I'd be interested in other's opinions about this, maybe it's just me, @MartinMinkov @bkase